src/dynamic_boundary_conditions/rainfall_data_from_hirds.py

```diff
@@ -98,10 +98,11 @@ def get_layout_structure_of_data(site_data: str) -> List[Tuple[int, float, str]]
     skip_rows = []
     rcp = []
     time_period = []
+    category = []
     # Read the site_data text string line by line with a for loop
     for index, line in enumerate(StringIO(site_data)):
         # Get lines that contain "(mm) ::" for depth data or "(mm/hr) ::" for intensity data
-        if (line.find("(mm) ::") != -1) or (line.find("(mm/hr) ::") != -1):
+        if re.search(r"\(mm\S*\)\s::", line):
```

Comment:
I like this re better than the previous one, but how would you feel about this?
Regular expressions can be quite hard to read but if this suggestion does not handle enough edge cases then the regular expression may be needed.

```suggestion
        if "(mm) ::" in line or "(mm/hr) ::" in line:
```

Reply:
Happy to change to the above suggestion (better than my previous one). I used this regex in case the unit '/hr/ changes. Let me know :)

Reply:
I'd be very surprised if '/hr' changes without everything else changing

Reply:
Changed.